pycloudlib/ibm/instance.py

```diff
@@ -746,9 +758,38 @@ def start(self, wait=True):
         if wait:
             self.wait()
 
+    def _check_instance_failed_status(self) -> None:
+        """
+        Check if the instance failed to start and raise an exception if so.
+
+        Raises:
+            IBMCapacityException: If the instance failed to start due to
+                capacity issues.
+            IBMException: If the instance failed to start due to other reasons.
+        """
+        if self._instance["status"] == _Status.FAILED.value:
+            if any(
+                "capacity" in reason["code"]
+                for reason in self._instance["status_reasons"]
+            ):
+                raise IBMCapacityException(
+                    f"Out of capacity! Instance {self.id} failed to start: "
+                    f"{self._instance['status_reasons'][0]['message']}"
+                )
+            # Raise generic IBM exception if not capacity related
+            raise IBMException(
+                f"Instance {self.id} failed to start: "
+                f"{self._instance['status_reasons'][0]['message']}"
+            )
+
     def _wait_for_instance_start(self, **kwargs):
         """Wait for the cloud instance to be up."""
-        self._wait_for_status(_Status.RUNNING)
+        # if self.b
+        self._wait_for_status(
+            _Status.RUNNING,
+            sleep_seconds=900,
```

Comment:
This is good for now. But, the `900` should be configurable from a consumer POV, as it might be not flexible enough.

Reply:
Okay I'll look at adding this sometime soon. Thanks!!